File: Project/Simulator/Organisms/Plants/Guarana.java
package Simulator.Organisms.Plants;

import Simulator.Organisms.Organism;

import org.json.JSONObject;

import Simulator.World;

public class Guarana extends Plant {

    public Guarana(World world) {
        super(0, world, Type.GUARANA);
    }

    public Guarana(JSONObject json, World world) {
        super(json, world);
    }

    @Override
    public Plant construct() {
        return new Guarana(world);
    }

    @Override
    public boolean collisionReaction(Organism other) {
        other.setPower(other.getPower() + 3);
        die();
        world.addLog(other.getSymbol() + " ate " + getSymbol() + " and gained 3 power!");
        return true;
    }
}

File: Project/Simulator/Organisms/Plants/Guarana.py
from .Plant import Plant
from Simulator.Organisms.Organism import Type, Organism


class Guarana(Plant):
    def __init__(self, world, json=None):
        super().__init__(0, world, Type.GUARANA, json)

    def construct(self):
        return Guarana(self.world)

    def collisionReaction(self, other: Organism):
        other.setPower(other.getPower() + 3)
        self.die()
        self.world.addLog(f"{other.getSymbol()} ate {self.getSymbol()} and gained 3 power!")
        return True
